cmake/DAGMC_macros.cmake

```diff
@@ -90,6 +90,10 @@ macro (dagmc_setup_options)
 
 if (DOUBLE_DOWN)
   find_package(DOUBLE_DOWN REQUIRED)
```

Comment:
Can you put the required version in the `find_package` command and let CMake take care of the error?

Reply:
I have done this in my previous commit. @pshriwise has come up with this idea. This will terminate the process if the minimum conditions aren't met with a custom message.

Reply:
I see... I wonder why @pshriwise doesn't want to just rely on CMake's error message?

Reply:
ping @pshriwise

Reply:
I think @gonuke is right. We should use the version argument of `find_package`.

Reply:
Ah, but now I'm remembering why we opted for this approach originally -- version ranges in `find_packages` are only supported since CMake 3.19 and our [CMakeLists.txt supports CMake back to 3.1](https://github.com/svalinn/DAGMC/blob/d9adb178ac97d19b423daff07c2bd731fbcc92cf/CMakeLists.txt#L2).

Reply:
So, if the desire is to specify a specific version of Double-Down then I'd vote to move to using `find_package` to do the version check, but if we really do want to support versions of Double-Down >= 1.1.0 then I think we need to keep the version check separate from the `find_package` call or update our minimum req'd CMake version.

Reply:
We are currently using `find_package` to set min version of Geant4 through PR #803 [here](https://github.com/svalinn/DAGMC/blob/50e5c8caa5990b27d66427dc69272302de3326f6/src/geant4/CMakeLists.txt#L3).

Reply:
Given that, CMake 3.1 is now quite old, I'm wondering if it's still capable of building the current build since we typically use Ubuntu 20.04 and 22.04 to build DAGMC in our workflow, which comes with a more recent version of CMake. Additionally, I've noticed a deprecation warning when using CMake 3.1, which suggests we upgrade to version 3.5.

Reply:
> We are currently using find_package to set min version of Geant4 through PR https://github.com/svalinn/DAGMC/pull/803 [here](https://github.com/svalinn/DAGMC/blob/50e5c8caa5990b27d66427dc69272302de3326f6/src/geant4/CMakeLists.txt#L3).

I believe that this use of the signature indicates that the Geant4 should _match_ the specified version, not a minimum required version. 

---

From the CMake documentation on `find_package`: 

The [version] argument requests a version with which the package found should be compatible. There are two possible forms in which it may be specified:

  - A single version with the format major[.minor[.patch[.tweak]]], where each component is a numeric value.

  - A version range with the format versionMin...[<]versionMax where versionMin and versionMax have the same format and constraints on components being integers as the single version. By default, both end points are included. By specifying <, the upper end point will be excluded. Version ranges are only supported with CMake 3.19 or later.

Reply:
@gonuke, what do you like to do? Should I consider updating CMake version to v3.19 from v3.1 (Ubuntu 20.04 uses [CMake v3.16.3](https://launchpad.net/ubuntu/+source/cmake/3.16.3-1ubuntu1)? Or I may change the current method of setting Geant4 min version?

Reply:
The Geant4 handling is interesting since we currently list a required version, but then handle versions newer than the required version and it works.  I guess I'm a little confused about what **_should_** work and what **_does_** work.

Reply:
Ubuntu 20.04 is failing as it uses CMake 3.16. Here is the [workflow](https://github.com/svalinn/DAGMC/actions/runs/7987368761/job/21809714439#step:7:1). I have set the DD version to 1.0.0 and it seems CMake is ignoring to check the minimum version of DD.